cooper/optim/constrained_optimizers/extrapolation_optimizer.py

```diff
@@ -58,13 +53,6 @@ def custom_sanity_checks(self):
                 Please ensure that all optimizers are extrapolation capable."""
             )
 
-        for multiplier in self.multipliers:
-            if getattr(multiplier, "restart_on_feasible", False):
-                raise RuntimeError(
-                    """Using restart on feasible for multipliers is not supported in
-                    conjunction with the ExtrapolationConstrainedOptimizer."""
-                )
-
     def step(self, call_extrapolation: bool = False):
```

Comment:
`@torch.no_grad`

Reply:
You mean for decorating `step`? 
I don't think that is correct for extrapolation. Remember that there is an extra ;) gradient computation inside this step, so you can't wrap this method in `no_grad`